### PR TITLE
[Drop-In UI] Updated the default navigation puck vector asset

### PR DESCRIPTION
**Pull request by tomaszrybakiewicz**
Closes NAVAND-921

### Description
- Added new vector asset for the default navigation puck 
   (`LocationPuckOptions.Builder.navigationPuck(context: Context)`)
- Added QA-Test-App “use legacy puck” toggle.
 
_NOTE: Use of legacy puck can be restored as following:_ 

```kotlin
navigationView.customizeViewStyles {
    locationPuckOptions = LocationPuckOptions.Builder(context)
        .defaultPuck(
            LocationPuck2D(
                bearingImage = ContextCompat.getDrawable(
                    context,
                    com.mapbox.navigation.ui.maps.R.drawable.mapbox_navigation_puck_icon,
                )
            )
        )
        .idlePuck(LocationPuckOptions.Builder.regularPuck(context))
        .build()
}
```
### Screenshots or Gifs
_QA-Test-App recording captured on Pixel 6_

https://user-images.githubusercontent.com/2678039/205340693-1f3e6e97-bc3a-4984-b4c2-c3107c12d1e0.mp4

### CODE REVIEW

File: libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/puck/LocationPuckOptions.kt
Comment (dzinad) [inline libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/puck/LocationPuckOptions.kt:215]: Can we use a more appropriate name? Like `mapbox_navigation_puck_icon_themed` or sth like that?

Comment (tomaszrybakiewicz) [reply libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/puck/LocationPuckOptions.kt:215]: Why? `mapbox_navigation_puck_icon2` says it all. Version 2 of `mapbox_navigation_puck_icon`.

**Comment by abhishek1508** [comment]
LGTM. @stnikita1 Can you please take a look and let us know if this is fine before we merge it?

File: libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/puck/LocationPuckOptions.kt
Comment (dzinad) [reply libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/puck/LocationPuckOptions.kt:215]: Because later we'll have `mapbox_navigation_puck_icon3`, `mapbox_navigation_puck_icon4`, `mapbox_navigation_puck_icon10`. How will we know which one stands for what? While "themed" describes it: the one that supports different themes.
But you've worked with UI and resources on the project much more that I did so no strong objections here, just a suggestion.

Comment (Zayankovsky) [reply libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/puck/LocationPuckOptions.kt:215]: Do we need the old drawable at all? Can we simply update mapbox_navigation_puck_icon instead of adding a new drawable?

Comment (tomaszrybakiewicz) [reply libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/puck/LocationPuckOptions.kt:215]: @dzinad We'll address this later if needed. `mapbox_navigation_puck_icon2` is a good enough name for this asset.

> Do we need the old drawable at all? Can we simply update mapbox_navigation_puck_icon instead of adding a new drawable?
@Zayankovsky We... no. Customers that already integrated Drop-In UI... yes. To allow those customers to switch back to the old puck easily, we should keep it. Besides, the asset is a vector graphic that weighs only 926 bytes.

**Comment by stnikita1** [comment]
@abhishek1508 can we also update the generic puck please? And I reduced the size of the guidance puck
Generic Puck
Guidance Puck

**Comment by abhishek1508** [comment]
@stnikita1 Thanks for adding the comment. 
@tomaszrybakiewicz Let us use the resized navigation puck. Looks like it's a bit smaller than the previous design.

@stnikita1  We will address the round generic puck to have a better shadow in a follow up PR.

**Comment by tomaszrybakiewicz** [comment]
> @stnikita1 Thanks for adding the comment. @tomaszrybakiewicz Let us use the resized navigation puck. Looks like it's a bit smaller than the previous design.
> 
> @stnikita1 We will address the round generic puck to have a better shadow in a follow up PR.

Done. I've update navigation puck to the one mentioned here https://github.com/mapbox/mapbox-navigation-android/pull/6678#issuecomment-1341421519, and changed it's fill color to `R.color.colorPrimary`. Please see attached screen recording.

https://user-images.githubusercontent.com/2678039/206537536-af018adb-f4ff-4fd2-a7bd-4c8196e08983.mp4

**Comment by abhishek1508** [comment]
@stnikita1 Can you take another look at it? The size of the puck has now been changed. I have shared the updated `.apk` on slack.

**Review by abhishek1508 — APPROVED** [review]
LGTM. We should wait for an approval from @stnikita1 before we merge

**Comment by stnikita1** [comment]
Just reviewed the last apk and the puck looks surprisingly the same size. Could we try a smaller object?
Guidance Puck

**Comment by abhishek1508** [comment]
> Just reviewed the last apk and the puck looks surprisingly the same size. Could we try a smaller object? Guidance Puck

The resized `.svg` that you shared [here](https://github.com/mapbox/mapbox-navigation-android/pull/6678#issuecomment-1341421519) had the following dimensions
```
android:width="61dp"
android:height="65dp"
android:viewportWidth="61"
android:viewportHeight="65"
```
The one you just shared in your last comment has the following dimensions

```
android:width="56dp"
android:height="59dp"
android:viewportWidth="56"
android:viewportHeight="59"
```

I just want to make sure that a value of 5px would make a difference. 

cc @stnikita1

**Comment by stnikita1** [comment]
@abhishek1508 The previous iteration wasn't radical too but we didnt notice any difference. So, maybe this time  it will work)

**Comment by stnikita1** [comment]
@tomaszrybakiewicz thank you for the screens. I think we definitely should stay with the puck B (61x65, stroke width 3) It's better visible on the map. I fix the size, now the container is square, and divide the layers (outline and object) to separate.
Btw do we change the color in day/night mode as well?
Guidance Puck

**Comment by abhishek1508** [comment]
> Btw do we change the color in day/night mode as well?

Yes we do. These are the colors we use as our primary color for Drop-In UI in day and night mode:
Day - #373674
Night - #4D58B5

We will be using the color above for the puck to maintain consistency

cc @stnikita1

**Comment by tomaszrybakiewicz** [comment]
@stnikita1 Updated. What do you think?


| New Nav Puck <br/>(64dp x 64dp)<br/>(day)<br/>| New Nav Puck<br/>(64dp x 64dp)<br/>(day) | Legacy Puck<br/>(75dp x 75dp)<br/>(day) | Legacy Puck<br/>(75dp x 75dp)<br/>(night) | 
|:---:|:---:|:---:|:---:|
|||||

**Comment by stnikita1** [comment]
@tomaszrybakiewicz thanks for the screenshots! It looks great. But the shadow is gone for some reason. I double check the svg, the shadow should be there

**Comment by tomaszrybakiewicz** [comment]
@stnikita1 I've mentioned it [in my previous comment](https://github.com/mapbox/mapbox-navigation-android/pull/6678#issuecomment-1343166087). Android doesn't support drop shadows in Vector Drawables. We'll have to figure something out and do this as a follow-up.

**Comment by tomaszrybakiewicz** [comment]
@stnikita1 Did some wizardry and applied a blur effect to the shadow asset using RenderScript.
Here's how it looks:
  | Day | Night |
  |:---:|:---:|
  |||

**Comment by stnikita1** [comment]
@tomaszrybakiewicz awesome! Thank you!

**Review by stnikita1 — APPROVED** [review]

**Pull request merged**
